include/powsybl/iidm/Variant.hpp

```diff
@@ -14,16 +14,22 @@ namespace powsybl {
 
 namespace iidm {
 
+class VariantManagerHolder;
+
 template <typename T>
 class Variant {
 public:
     virtual ~Variant() noexcept = default;
 
     virtual std::unique_ptr<T> copy() const = 0;
+
+    virtual void setVariantManagerHolder(VariantManagerHolder& variantManagerHolder);
```

Comment:
Default definition to update ref to `VariantManagerHolder` (necessary because `VariantArray` is generic.
Does nothing for `powsybl::iidm::bus_breaker_voltage_level::VariantImpl` and `powsybl::iidm::node_breaker_voltage_level::VariantImpl` and overriden by `powsybl::iidm::network::VariantImpl`.